torch/fx/experimental/graph_gradual_typechecker.py

```diff
@@ -0,0 +1,114 @@
+from functools import reduce
+import torch
+from torch.fx.tensor_type import Dyn, is_consistent, TensorType, is_more_precise
+import operator
+
+
+class GraphTypeChecker:
+    def __init__(self, env, traced):
+        self.env = env
+        self.traced = traced
+
+    def type_check(self):
+        """
+        A gradual type checker for graphs
+        Effect: every node's field type will be
+        populated with a type after type-checking is done
+        """
+        # here we collect types for variables
+        graph = self.traced.graph
+
+        # type check every node with gradual type rules
+        # if any node does not type check return false
+        for n in graph.nodes:
+            if not self.type_check_node(n):
+                return False
+        return True
+
+    def type_check_node(self, n):
```

Comment:
To help improve readability, can you add type annotations to the function parameters here and on collect_params, as well as add a short comment describing what is expected for each of the arguments?

Reply:
collect_params is removed now, clarified what n is in a comment, however if I annotate it with Node, mypy fails with 

`error: Item "Tuple[Any, ...]" of "Union[Tuple[Any, ...], List[Any], Dict[str, Any], slice, Node, str, int, float, bool, dtype, Tensor, device, None]" has no attribute "type"`

is it possible that not all nodes have the type attribute?